# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/actors/SubframeCrashChild.jsm
# Commit: 2d0e5336e455
# Full Hash: 2d0e5336e455f560cdd345afa57f21b8cdbfc1fb
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2019-08-26 22:30:36
# Regressor Bug: 1563619
# File Overlap Count: 1
# Description:
#   Bug 1563619 - Handle subframe crashes in BrowserParent::ActorDestroy. r=mconley,kmag,peterv
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D40937
# ==============================================================================

diff -r fd3185644de9 -r 2d0e5336e455 browser/actors/SubframeCrashChild.jsm
--- a/browser/actors/SubframeCrashChild.jsm	Mon Aug 26 13:07:26 2019 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,30 +0,0 @@
-/* vim: set ts=2 sw=2 sts=2 et tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-"use strict";
-
-var EXPORTED_SYMBOLS = ["SubframeCrashChild"];
-
-let { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
-
-class SubframeCrashChild extends JSWindowActorChild {
-  receiveMessage(message) {
-    if (message.name == "SubframeCrashed") {
-      this.onSubframeCrashed(message.data.id);
-    }
-  }
-
-  onSubframeCrashed(browsingContextId) {
-    let bc = BrowsingContext.get(browsingContextId);
-    let iframe = bc.embedderElement;
-    let uri = Services.io.newURI(iframe.src);
-    iframe.removeAttribute("fission");
-    iframe.removeAttribute("src");
-    // Passing a null remoteType is currently how we tell the iframe to run
-    // in the same process as the embedder.
-    iframe.changeRemoteness({ remoteType: null });
-    let docShell = iframe.frameLoader.docShell;
-    docShell.displayLoadError(Cr.NS_ERROR_FRAME_CRASHED, uri, null);
-  }
-}